specutils/fitting/spline_continuum.py

```diff
@@ -0,0 +1,485 @@
+        The type of model to use for the continuum.
+        Must be astropy.modeling.FittableModel
+        See astropy.modeling.models
+        Default: models.Chebyshev1D(3)
+        TODO add a spline option (since this is not currently implemented)
+    fitter : `astropy.modeling.fitting.Fitter`
+        The type of fitter to use for the continuum.
+        See astropy.modeling.fitting for valid choices
+        TODO currently does not typecheck because fitters do not subclass fitting.Fitter
+        Default: fitting.LevMarLSQFitter()
+    sigma : float, optional
+        The number of standard deviations to use for both lower and upper clipping limit.
+        Defaults to 3.0
+    sigma_lower : float or None, optional
+        Number of standard deviations for lower bound clipping limit.
+        If None (default), then `sigma` is used.
+
+    sigma_upper : float or None, optional
+        Number of standard deviations for upper bound clipping limit.
+        If None (default), then `sigma` is used.
+    iters : int or None, optional
+        Number of iterations to perform sigma clipping.
+        If None, clips until convergence achieved.
+        Defaults to 5
+    exclude_regions : list of tuples, optional
+        A list of dispersion regions to exclude.
+        Each tuple must be sorted.
+        e.g. [(6555,6575)]
+    full_output : bool, optional
+        If True, return more information.
+        Currently, just the model and the pixels-used boolean array
+
+    Returns
+    -------
+    continuum_model : `astropy.modeling.FittableModel`
+        Output a model for the continuum
+
+    Raises
+    ------
+    ValueError
+       If: spectrum is not the correct type,
+       the exclude regions do not satisfy a list of sorted tuples,
+       the model and/or fitter are of the wrong type,
+
+    Examples
+    --------
+    TODO: add more and unit tests
+
+    See https://github.com/spacetelescope/dat_pyinthesky/blob/master/pyinthesky_specutils_fitting.ipynb
+
+    """
+
+    # Parameter checks
+    if not isinstance(spectrum, Spectrum1D):
+        raise ValueError('The spectrum parameter must be a Spectrum1D object')
+
+    exclude_regions = [] if exclude_regions is None else exclude_regions
+
+    for exclude_region in exclude_regions:
+        if len(exclude_region) != 2:
+            raise ValueError('All exclusion regions must be of length 2')
+        if exclude_region[0] >= exclude_region[1]:
+            raise ValueError('All exclusion regions must be (low, high)')
+
+    # Set default model and fitter
+    if model is None:
+        logging.info("Using Chebyshev1D(3) as default continuum model")
+        model = models.Chebyshev1D(3)
+
+    if fitter is None:
+        fitter = fitting.LevMarLSQFitter()
+
+    if not isinstance(model, modeling.FittableModel):
+        raise ValueError('The model parameter must be a astropy.modeling.FittableModel object')
+    # TODO: this is waiting on a refactor in modeling.fitting to work
```

Comment:
@alexji can you expand on what specifically this is waiting for in the refactor?

Reply:
I can't remember exactly, but I believe it had to do with the fact that at the time astropy.modeling did not support any type of spline fitting.